src/coreComponents/common/format/table/TableLayout.hpp

```diff
@@ -51,68 +55,328 @@ class TableLayout
+     * @param size The size to set as the maximum string length.
+     */
+    void setMaxStringSize( size_t const size );
+
+    /**
+     * @brief Gets the maximum string size of the column.
+     * @return size_t The maximum string size of the column.
+     */
+    size_t getMaxStringSize() const;
+
+    /**
+     * @brief Adds multiple sub-columns to the column.
+     * @param subColName A list of sub-column names to add.
+     * @return The current column object
+     */
+    TableLayout::Column & addSubColumns( std::initializer_list< TableLayout::Column > subCol );
+
+    /**
+     * @brief Adds multiple sub-columns to the column.
+     * @param subColName A list of sub-column names to add.
+     * @return The current column object
+     */
+    TableLayout::Column & addSubColumns( std::initializer_list< string > subColName );
+
+    /**
+     * @brief Adds a single sub-column to the column.
+     * @param subColName The name of the sub-column to add.
+     * @return The current column object.
+     */
+    TableLayout::Column & addSubColumns( string const & subColName );
+
+    /**
+     * @brief Sets the header alignment for the column.
+     * @param headerAlignment The alignment to set for the column header (left, right, or center).
+     * @return The current column object
+     */
+    TableLayout::Column & setHeaderAlignment( Alignment headerAlignment );
+
+    /**
+     * @brief Sets the values alignment for the column.
+     * @param headerAlignment The alignment to set for the column values (left, right, or center).
+     * @return The current column object
+     */
+    TableLayout::Column & setValuesAlignment( Alignment valueAlignment );
+
+    /**
+     * @brief Checks if the column has any child columns.
+     * @return bool True if the column has child columns, otherwise false.
+     */
+    bool hasChild() const;
+
+    /**
+     * @brief Checks if the column has a parent column.
+     * @return bool True if the column has a parent, otherwise false.
+     */
+    bool hasParent() const;
+
+private:
+    /// The maximum string size of the column.
+    size_t m_maxStringSize;
+  };
+
+  /**_est columns / sub columns.
+   * An exemple of an iteration: A -> B.A -> B.B -> B.C -> C.A.A -> C.A.B -> C.B.A -> C.B.B -> D
    */
-  struct Column
+  class LeafIterator
   {
-    /// Structure who contains parameters for a column
-    ColumnParam m_parameter;
-    /// A vector containing all column values
-    std::vector< string > m_columnValues;
-    /// The largest string in the column
-    string m_maxStringSize;
+public:
+    using ColumnType = Column;
+
+    LeafIterator( ColumnType * columnPtr, size_t idxLayer ):
+      m_currentColumn( columnPtr ), m_currentLayer( idxLayer )
+    {}
+
+
+    LeafIterator & operator=( Column * columnPtr )
+    {
+      this->m_currentColumn= columnPtr;
+      return *this;
+    }
+
+    // Prefix ++ overload
+    LeafIterator & operator++()
```

Comment:
*Only one-liners in headers* -> please move definition to `.cpp`.
There are a lot more in this header after this line.

By one liner I mean:
```
    size_t getCurrentLayer() const
    { return m_currentLayer; }

    LeafIterator( ColumnType * columnPtr, size_t idxLayer ):
      m_currentColumn( columnPtr ), m_currentLayer( idxLayer )
    {}
```

and not
```
    size_t getCurrentLayer() const
    {
      return m_currentLayer;
    }
```

And long templated methods go at the bottom of the header.